### PR TITLE
Pipe: clear reference count for events in retry queue & detect re-increasing reference count for other connectors

### PR DESCRIPTION
**Pull request by VGalaxies**
follow up #12315

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/event/EnrichedEvent.java [hunk 1]
```diff
@@ -148,6 +152,9 @@ public boolean decreaseReferenceCount(String holderMessage, boolean shouldReport
             Thread.currentThread().getStackTrace());
       }
     }
+    if (!isSuccessful) {
```
Comment (Caideyipi) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/event/EnrichedEvent.java:155]: It seems that the flag is meaningless and the "0" reference count will still go "<0" and succeed.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/connector/PipeReceiverStatusHandler.java [hunk 2]
```diff
@@ -133,12 +133,14 @@ public void handle(TSStatus status, String exceptionMessage, String recordMessag
 
           LOGGER.warn(
               "User conflict exception: will retry {}. status: {}",
-              (retryMaxMillisWhenConflictOccurs == Long.MAX_VALUE ? "forever" : "for at least ")
-                  + (retryMaxMillisWhenConflictOccurs
-                          + exceptionFirstEncounteredTime.get()
-                          - System.currentTimeMillis())
-                      / 1000.0
-                  + " seconds",
+              retryMaxMillisWhenConflictOccurs == Long.MAX_VALUE
```
Comment (Caideyipi) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/connector/PipeReceiverStatusHandler.java:136]: Personally I think keeping the "()" is clearer...

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/event/EnrichedEvent.java (hunk 1 above)
Comment (VGalaxies) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/event/EnrichedEvent.java:155]: The `isSuccessful` flag is used to determine whether internal increases or decreases were successful. In the case where the reference count is less than zero, we still consider the decrease to be successful.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/connector/PipeReceiverStatusHandler.java (hunk 2 above)
Comment (VGalaxies) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/connector/PipeReceiverStatusHandler.java:136]: The previous approach would result in appending "xxx seconds" after "forever".

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/sync/IoTDBDataRegionSyncConnector.java [hunk 3]
```diff
@@ -182,6 +224,8 @@ private void doTransfer() {
           tabletBatchBuilder.deepCopyEvents().toString());
     }
 
+    tabletBatchBuilder.decreaseEventsReferenceCount(
+        IoTDBDataRegionSyncConnector.class.getName(), false);
```
[the remark this answers is not in the document]
Comment (VGalaxies) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/sync/IoTDBDataRegionSyncConnector.java:228]: fixed

[the remark this answers is not in the document]
Comment (SteveYurongSu) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/sync/IoTDBDataRegionSyncConnector.java:228]: > Why false here?

Nice catch! @Caideyipi

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**